src/chapters/http/how-the-internet-works.rst

```diff
@@ -0,0 +1,193 @@
+to send it? Also, once it receives a request, how does the server know where to
+send the data?
+
+.. index:: ! IP address
+
+Every device connected directly or indirectly to the internet has its own
+numerical address, called an **IP address**. "IP" stands for *internet
+protocol*. The links we click on a webpage contain IP addresses, which identify
+the server we need to contact. Requests also include the address of our device.
+
+Any time our request passes through a different device on its way to the server,
+that device adds its own IP address to the request. Think of this like a trail
+of breadcrumbs, or as a series of directions on a map. When the server responds
+to our request, it sends the data through the same IP addresses, but in reverse
+order.
+
+.. figure:: figures/ip-trail.png
+   :alt: A series of IP addresses are gathered as a request makes its way to the server.
+   :width: 80%
+
+   IP addresses provide a path for data to follow.
+
+DNS
+^^^
+
+IP addresses look something like this: ``109.203.100.231``. When we surf the
+web, however, these addresses are usually hidden. Instead, we see names like
+``google.com``, which are easier to understand.
+
+.. index:: ! DNS
+
+**DNS** is the address book of the internet. It provides readable and memorable
+names for servers, such as ``launchcode.org``. The are called **domain names**,
+and they stand in for the actual server IP addresses.
+
+.. admonition:: Example
+
+   The domain names for ``104.25.127.113`` and ``172.217.8.5`` are
+   ``launchcode.org`` and ``mail.google.com``, respectively.
+
+.. admonition:: Try It!
+
+   It's easy to look up the IP address of any domain name using some free
+   online tools. 
+
+   Use the popular site `MX Toolbox <https://mxtoolbox.com/DNSLookup.aspx>`__
+   to look up the IP address for ``python.org``.
+
+Protocols
+---------
+
+Once a client or server knows where to send a request/response, the actual data
+needs to get transferred.
+
+.. index::
+   single: url; protocol
+
+A **protocol** is a set of rules for communicating between computers. Most web
+communication uses three protocols. We've looked at one already, DNS, and here
+is a summary of the other two.
+
+.. list-table:: Common Web Protocols
+   :header-rows: 1
+
+   * - Protocol
+     - Full Name
+     - Role
+   * - HTTP
+     - Hypertext Transfer Protocol
+     - Used to transfer files and information, including:
+
+       - HTML, CSS, and JavaScript files
+       - Images and other media
+       - Form submissions
+
+   * - TCP/IP
+     - Transmission Control Protocol / Internet Protocol
+     - Used to transfer small chunks of raw data known as *packets*.
+   * - DNS
+     - Domain Name Service
+     - Translates human-friendly names into server addresses.
+
+A complete discussion for each of these protocols is beyond the scope of this
+class. However, we should recognize how each one plays a role in making web
+communication happen.
+
+HTTP
```

Comment:
This isn't super important, but I feel like diagrams could help these two protocol sections be a bit easier to grasp. Maybe something to consider in the future, if not now...

Reply:
I added a diagram for TCP/IP to the repo.  I've got another one for HTTP vs. HTTPS, but you should look at that one first, I think.